src/components/codecSupportHelper.ts

```diff
@@ -1,7 +1,196 @@
+    }
+}
+
+/**
+ * Get the string to use for testing support of a codec.
+ * @param codec - The codec in question.
+ * @param profile - The profile for the codec.
+ * @param level - The level for the codec.
+ * @param bitDepth - The bit depth of the video.
+ * @returns The string to use for testing support of the codec.
+ */
+function getCodecString(
+    codec: VideoCodec,
+    profile?: string,
+    level?: number,
+    bitDepth?: number
+): string {
+    switch (codec) {
+        case VideoCodec.H264: {
+            // Default to the oldest baseline profile.
+            profile = profile ?? 'baseline';
+
+            let profileFlag: string;
+
+            switch (profile) {
+                case 'high 10':
+                    profileFlag = '6e00';
+                    break;
+                case 'high':
+                    profileFlag = '6400';
+                    break;
+                case 'main':
+                    profileFlag = '4d00';
+                    break;
+                case 'baseline':
+                    profileFlag = '4200';
+                    break;
+                case 'constrained baseline':
+                    profileFlag = '4240';
+                    break;
+                default:
+                    profileFlag = '4200';
+                    break;
+            }
+
+            // Levels are bound by max frame size (macroblocks) and decoding
+            // speed (macroblocks/s).
+            // A macroblock is 16x16 pixels.
+            //
+            // See:
+            //   * https://en.wikipedia.org/wiki/Advanced_Video_Coding#Levels
+            level = level ?? 10;
+
+            return `avc1.${profileFlag}${level.toString(16)}`;
+        }
+        case VideoCodec.H265: {
+            let profileFlag: string;
+            let constraintFlag: number;
+
+            switch (profile) {
+                case 'main':
+                    profileFlag = 'L';
+                    constraintFlag = 0;
+                    break;
+                case 'main 10':
+                    profileFlag = 'L';
+                    constraintFlag = 4;
+                    break;
+                case 'high':
+                    profileFlag = 'H';
+                    constraintFlag = 4;
+                    break;
+                case 'high 10':
+                    profileFlag = 'H';
+                    constraintFlag = 4;
+                    break;
+                default:
+                    profileFlag = 'L';
+                    constraintFlag = 0;
+                    break;
+            }
+
+            // Levels are bound by the luma picture size (total pixels) and
+            // luma sample rate (samples/s).
+            level = level ?? 10;
+
+            return `hev1.1.${constraintFlag}.${profileFlag}${level * 3}.B0`;
+        }
+        case VideoCodec.VP8:
+            return 'vp8';
+        case VideoCodec.VP9: {
+            let profileFlag: string;
+
+            switch (profile) {
```

Comment:
Might we want a `.toLowerCase()` here for defensive purposes (and change the case literals correspondingly)?